sdk/tables/azure-data-tables/azure/data/tables/_table_batch.py

```diff
@@ -238,11 +238,19 @@ def update(
         self._verify_partition_key(entity)
         temp = entity.copy()
 
+        match_condition = kwargs.pop("match_condition", None)
+        etag = kwargs.pop("etag", None)
+        if match_condition and not etag:
+            try:
+                etag = entity.metadata.get("etag", None)
+            except AttributeError:
```

Comment:
I think you might also need to catch `TypeError` here - if a customer passes in a `TableEntity` instance they created, then `entity.metadata` will be `None` (and we should add a test for this).

Reply:
Updated and added a test!